cli/lib/kontena/cli/stacks/yaml/reader.rb

```diff
@@ -368,7 +368,12 @@ def interpolate(content, use_opto: true, substitutions: {}, raise_on_unknown: fa
               if use_opto
                 opt = variables.option(var)
                 if opt.nil?
-                  raise RuntimeError, "Undeclared variable '#{var}' in #{file}:#{line_num} -- #{row}" if raise_on_unknown
+                  to_env = variables.find { |opt| opt.to[:env][var] }
+                  if to_env
+                    val = to_env.value
+                  else
+                    raise RuntimeError, "Undeclared variable '#{var}' in #{file}:#{line_num} -- #{row}" if raise_on_unknown
+                  end
                 else
                   val = opt.value
                   if opt.to['env']
```

Comment:
What does this `if opt.to['env']` => `Array(opt.to['env']).each { |env_var| env[env_var] = val.to_s }` do? Doesn't the `to: env` resolver itself already do that?

Reply:
It indeed should. I don't know why it's there